src/backend/query_builder.rs

```diff
@@ -1215,13 +1214,22 @@ pub trait QueryBuilder: QuotedBuilder + EscapeBuilder + TableRefBuilder {
     /// Translate part of a condition to part of a "WHERE" clause.
     fn prepare_condition_where(&self, condition: &Condition, sql: &mut dyn SqlWriter) {
         if condition.negate {
-            write!(sql, "NOT (").unwrap();
+            write!(sql, "NOT ").unwrap()
         }
-        let mut is_first = true;
-        for cond in &condition.conditions {
-            if is_first {
-                is_first = false;
-            } else {
+
+        if condition.is_empty() {
+            match condition.condition_type {
+                ConditionType::All => self.prepare_constant(&true.into(), sql),
+                ConditionType::Any => self.prepare_constant(&false.into(), sql),
+            }
+            return;
+        }
```

Comment:
Sorry for the coming late. I wonder why they are different?
- `Cond::all()` => `TRUE`
- `Cond::any()` => `FALSE`

Shouldn't it both be `FALSE` to prevent unintentional empty condition?

Reply:
They should each reduce to the identity element of the operation: See https://en.wikipedia.org/wiki/Boolean_algebra#Monotone_laws .

Basically, `all(a, b, c)` is equivalent to `all(true, a, b, c)`, so `all()` should be `all(true)`. Same for `any` and `false`.

In other terms, think about "I want to delete if _any_ of these conditions are true" (and you didn't add conditions) -> You don't want to delete.

Reply:
👍